packages/camera/camera_android_camerax/lib/src/camera_info.dart

```diff
@@ -0,0 +1,98 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart' show BinaryMessenger;
+
+import 'camerax.pigeon.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+
+/// Represents the metadata of a camera.
+class CameraInfo extends JavaObject {
+  /// Constructs a [CameraInfo] object.
+  CameraInfo.detached(
+      {BinaryMessenger? binaryMessenger, InstanceManager? instanceManager})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    setUpCameraInfoApis(binaryMessenger, instanceManager);
```

Comment:
The pattern of setting the `FlutterApi` only when a constructor is called can lead to a bug later. For instance, what happens in a scenario where a method from another class returns a `CameraInfo` before this constructor is called. In WebView, we would put the `FlutterApis` from every class into a single class: https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_android/lib/src/android_webview_api_impls.dart#L36

Alternatively, you could also set the `FlutterApi`s in `AndroidCameraCameraX. registerWith`.